metis-core/metis-core-service/src/test/java/eu/europeana/metis/core/service/TestRedirection.java

```diff
@@ -0,0 +1,269 @@
+        indexToPreviewPlugin.setPluginStatus(PluginStatus.FINISHED);
+        indexToPreviewPlugin.setStartedDate(Date.from(Instant.now().minus(90, ChronoUnit.MINUTES)));
+        indexToPreviewPlugin.setFinishedDate(Date.from(Instant.now().minus(60, ChronoUnit.MINUTES)));
+        indexToPreviewPlugin.setDataStatus(DataStatus.VALID);
+        return indexToPreviewPlugin;
+    }
+
+    @NotNull
+    private static IndexToPublishPlugin getIndexToPublishPlugin(IndexToPublishPluginMetadata indexToPublishPluginMetadata,
+                                                                ExecutionProgress executionProgress) {
+        final IndexToPublishPlugin indexToPublishPlugin = (IndexToPublishPlugin)
+                ExecutablePluginFactory.createPlugin(indexToPublishPluginMetadata);
+        indexToPublishPlugin.setExecutionProgress(executionProgress);
+        indexToPublishPlugin.setPluginStatus(PluginStatus.FINISHED);
+        indexToPublishPlugin.setStartedDate(Date.from(Instant.now().minus(45, ChronoUnit.MINUTES)));
+        indexToPublishPlugin.setFinishedDate(Date.from(Instant.now().minus(30, ChronoUnit.MINUTES)));
+        indexToPublishPlugin.setDataStatus(DataStatus.VALID);
+        return indexToPublishPlugin;
+    }
+
+    @NotNull
+    private static ExecutionProgress getExecutionProgress() {
+        final ExecutionProgress executionProgress = new ExecutionProgress();
+        executionProgress.setStatus(TaskState.PROCESSED);
+        executionProgress.setExpectedRecords(1);
+        executionProgress.setProcessedRecords(1);
+        executionProgress.setTotalDatabaseRecords(1);
+        return executionProgress;
+    }
+
+    @NotNull
+    private static HTTPHarvestPluginMetadata getHttpHarvestPluginMetadata() {
+        final HTTPHarvestPluginMetadata httpHarvestPluginMetadata = new HTTPHarvestPluginMetadata();
+        httpHarvestPluginMetadata.setUrl("http://url.org");
+        httpHarvestPluginMetadata.setUser("user");
+        httpHarvestPluginMetadata.setIncrementalHarvest(false);
+        httpHarvestPluginMetadata.setEnabled(true);
+        return httpHarvestPluginMetadata;
+    }
+
+    @NotNull
+    private static IndexToPreviewPluginMetadata getIndexToPreviewPluginMetadata() {
+        final IndexToPreviewPluginMetadata indexToPreviewPluginMetadata = new IndexToPreviewPluginMetadata();
+        indexToPreviewPluginMetadata.setIncrementalIndexing(false);
+        indexToPreviewPluginMetadata.setHarvestDate(Date.from(Instant.now().minus(100, ChronoUnit.MINUTES)));
+        indexToPreviewPluginMetadata.setPreserveTimestamps(false);
+        indexToPreviewPluginMetadata.setEnabled(true);
+        indexToPreviewPluginMetadata.setDatasetIdsToRedirectFrom(List.of());
+        indexToPreviewPluginMetadata.setPerformRedirects(true);
+        return indexToPreviewPluginMetadata;
+    }
+
+    @NotNull
+    private static Dataset getTestDataset(String datasetId) {
```

Comment:
Is the parameter datasetId required since it's always the same value?

Reply:
yes. removed.